fix(authz): include realm identifier in OPA authorization input for tenant isolation (#4992)
apache/polaris: CHANGELOG.md

### Deprecations

### Fixes

> - OPA authorizer now includes the realm identifier in the authorization context sent to OPA (`input.context.realm`). This ensures OPA policies can enforce tenant isolation across realms (previously the OPA input was identical for the same principal/role/resource names in different realms, unlike the Ranger authorizer).

Reviewer: Nit: I think the wording `unlike the Ranger authorizer` is unnecessary and maybe a bit misleading. 
```suggestion
- OPA authorizer now includes the realm identifier in the authorization context sent to OPA (`input.context.realm`). This ensures OPA policies can enforce tenant isolation across realms, preventing potential collisions if identical principal or resource names exist in different realms..
```

- The token broker now builds the JWT `Algorithm` and `JWTVerifier` once per realm in the `TokenBrokerFactory` and reuses them across requests, instead of rebuilding them on every `verify()`/`sign()` call on the request-scoped broker. For deployments using file-based symmetric secrets, the secret is now read once per realm (at first use) rather than on every JWT operation; rotating the on-disk secret requires a restart.

### Fixes

> - OPA authorizer now includes the realm identifier in the authorization context sent to OPA (`input.context.realm`). This ensures OPA policies can enforce tenant isolation across realms, preventing potential collisions if identical principal or resource names exist in different realms.

Reviewer 2: This is under the `1.6.0` section, but the PR targets `main`... This should probably be in the `Unreleased` section.
